units/event_notifier.go

```diff
@@ -0,0 +1,252 @@
+	q        amboy.Queue
+	env      evergreen.Environment
+	flags    *evergreen.ServiceFlags
+
+	Events []string `bson:"events"`
+}
+
+func makeEventNotifierJob() *eventNotifierJob {
+	j := &eventNotifierJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    eventNotifierName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewEventNotifierJob(env evergreen.Environment, q amboy.Queue, id string, events []string) amboy.Job {
+	j := makeEventNotifierJob()
+	j.q = q
+	j.env = env
+
+	j.SetID(fmt.Sprintf("%s:%s", eventNotifierName, id))
+	j.Events = events
+
+	return j
+}
+
+func (j *eventNotifierJob) Run(ctx context.Context) {
+	if len(j.Events) == 0 {
+		return
+	}
+	if j.q == nil {
+		j.q = j.env.RemoteQueue()
+	}
+	if j.q == nil || !j.q.Info().Started {
+		j.AddError(errors.New("evergreen environment not setup correctly"))
+		return
+	}
+	var err error
+	j.flags, err = evergreen.GetServiceFlags()
+	if err != nil {
+		j.AddError(errors.Wrap(err, "error retrieving admin settings"))
+		return
+	}
+	if j.flags.EventProcessingDisabled {
+		grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
+			"job":     eventMetaJobName,
+			"message": "events processing is disabled",
+		})
+		return
+	}
+
+	events, err := event.FindEventsByIDs(j.Events)
+	if err != nil {
+		j.AddError(err)
+		return
+	}
+
+	j.AddError(j.dispatchLoop(ctx, events))
+}
+
+func (j *eventNotifierJob) dispatchLoop(ctx context.Context, events []event.EventLogEntry) error {
+	// TODO: in the future we may want to split up this job so
+	// that the parallelism is pushed up to amboy rather than down
+	// into this job.
+	startTime := time.Now()
+	logger := event.NewDBEventLogger(event.AllLogCollection)
+	catcher := grip.NewSimpleCatcher()
+
+	input := make(chan event.EventLogEntry, len(events))
+	for _, e := range events {
+		select {
+		case input <- e:
+			continue
+		case <-ctx.Done():
+			return errors.New("operation aborted")
+		}
+	}
+	close(input)
+
+	wg := &sync.WaitGroup{}
+	for i := 0; i < runtime.NumCPU(); i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			defer recovery.LogStackTraceAndContinue("problem during notification dispatch")
+
+			for e := range input {
+				n, err := j.tryProcessOneEvent(&e)
+				catcher.Add(err)
+				catcher.Add(logger.MarkProcessed(&e))
+
+				if err = notification.InsertMany(n...); err != nil {
+					grip.Error(message.WrapError(err, message.Fields{
+						"job_id":        j.ID(),
+						"job":           eventMetaJobName,
+						"source":        "events-processing",
+						"notifications": n,
+						"message":       "can't insert notifications",
+					}))
+					catcher.Add(err)
+				}
+
+				catcher.Add(dispatchNotifications(ctx, n, j.q, j.flags))
+			}
+		}()
+	}
+	wg.Wait()
+
+	endTime := time.Now()
+	totalDuration := endTime.Sub(startTime)
+
+	grip.Info(message.Fields{
+		"job_id":     j.ID(),
+		"job":        eventMetaJobName,
```

Comment:
j.Type().Name rather than the const, and maybe change job to job_type

Reply:
done